.github/workflows/test-build-deploy.yml

```diff
@@ -80,10 +82,97 @@ jobs:
           mkdir /tmp/images
           ln -s /tmp/images ./docker-images
           make BUILD_IN_CONTAINER=false save-images
-      - name: Zip Images
-        run: tar -zcvf images.tar.gz /tmp/images
-      - name: Upload Images Artifact
+      - name: Create Docker Images Archive
+        run: tar -cvf images.tar.gz /tmp/images
+      - name: Upload Docker Images Artifact
         uses: actions/upload-artifact@v2
         with:
           name: Docker Images
           path: ./images.tar.gz
+
+  integration:
+    needs: build
+    runs-on: ubuntu-16.04
+    steps:
+      - name: Install Docker Client
```

Comment:
This is also done in `build`. Is there any way to build a shared step? Just asking, not a blocker.

Reply:
We could replace this with a YAML anchor (which GitHub Actions has plans to release [shortly](https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851/14)) or we could write a [custom action](https://github.com/actions/runner/issues/646) and use that instead. With the custom action we would replace the `Install Docker Client` step with something like this in both Build and Integration.

```yaml
steps:
  - name: Install Docker Client
     uses: cortexproject/install-docker-client
  - run: ...
``` 

This means you'll have to maintain another actions repo and that might bring unwanted toil when upgrading versions.

Reply:
A simple way to have this logic in one place would be to create an install-docker.sh script and call it whenever an image requires a clean docker client installation. This would also make it easier to update the docker version in a single place. Is that close to what you had in mind?

Reply:
^ This is a better idea imo. This way you won't need to update the docker client version manually every time the `install docker client` step is called. The script can simply read `$VER` which can be set as a global variable for the entire workflow at the top of the file.

Reply:
I committed the solution using an install-docker script. Let me know if it is okay. @pracucci